### PR TITLE
fix the url init in credentials

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/595

### CODE REVIEW

File: sdk/core/azure-core/src/credentials/credentials.cpp [hunk 1]
```diff
@@ -49,8 +49,9 @@ AccessToken Azure::Core::Credentials::ClientSecretCredential::GetToken(
   static std::string const errorMsgPrefix("ClientSecretCredential::GetToken: ");
   try
   {
-    Http::Url url;
-    url.SetPath(m_authority + UrlEncode(m_tenantId) + "/oauth2/v2.0/token");
+    Http::Url url(m_authority);
+    url.AppendPath(UrlEncode(m_tenantId));
```
Comment (Jinming-Hu) [inline sdk/core/azure-core/src/credentials/credentials.cpp:53]: maybe don't need to encode

Comment (antkmsft) [reply sdk/core/azure-core/src/credentials/credentials.cpp:53]: .NET MSAL does it, but differently. And in most cases, encoded and non-encoded tenant ID look the same, if we are talking about the ones you can realistically have. I'd say, at this point whether you do `UrlEncode(m_tenantId)` or not, it shouldn't matter much. Previously, it was working with us doing the encoding. But on the other hand, no code was using it at a time. Plus, it is always easier to not encode, and tell the client code to encode.

**Review by antkmsft — APPROVED** [review]

File: sdk/core/azure-core/src/credentials/credentials.cpp [hunk 2]
```diff
@@ -49,8 +49,9 @@ AccessToken Azure::Core::Credentials::ClientSecretCredential::GetToken(
   static std::string const errorMsgPrefix("ClientSecretCredential::GetToken: ");
   try
   {
-    Http::Url url;
-    url.SetPath(m_authority + UrlEncode(m_tenantId) + "/oauth2/v2.0/token");
+    Http::Url url(m_authority);
```
Comment (ahsonkhan) [inline sdk/core/azure-core/src/credentials/credentials.cpp:52]: Tests?

Comment (vhvb1989) [reply sdk/core/azure-core/src/credentials/credentials.cpp:52]: entire feature (Token Credential policy) is only covered with e2e tests from Storage for now.

**Review by ahsonkhan — APPROVED** [review]

**Pull request merged**